src/main/java/duke/TaskManager.java

```diff
@@ -0,0 +1,92 @@
+package duke;
+
+import duke.exception.TaskIndexOutOfBound;
+import duke.exception.EmptyDoneIndexException;
+
+import java.util.ArrayList;
+
+public class TaskManager {
+    public static final int MAX_TASKS = 100;
+    public  ArrayList<Task> tasks = new ArrayList<>();
+
+    public void addTask(Task task){
+        tasks.add(task);
+        System.out.println("Got it. I've added this task: ");
+        System.out.println("  " + task.getTask());
+        System.out.println("Now you have " + tasks.size() + " tasks on the list.");
+    }
+
+    public boolean searchTask(Task task){
+        boolean taskFound = false;
+        for(int i=0; i<tasks.size(); i++){
+            if(tasks.get(i).getTask().equals(task.getTask()) && !tasks.get(i).isDone()){   //be aware that maybe it is added but it is done, so your can add another one
+                taskFound = true;
+            }
+        }
+        return taskFound;
+    }
+
+    public void deleteTask(String taskDelete) throws TaskIndexOutOfBound,EmptyDoneIndexException {
+        int indexOfTask = getIndexOfTask(taskDelete);
+        Task taskToDelete = tasks.get(indexOfTask - 1);
+        tasks.remove(indexOfTask - 1);
+        System.out.println("Noted! I've removed this task: ");
+        System.out.println(taskToDelete.getTask());
+        System.out.println("Now you have" +  tasks.size()  + "tasks in the list.");
+    }
+
+    public void tasksDone(String taskDone) throws TaskIndexOutOfBound,EmptyDoneIndexException {
+        int indexOfTask = getIndexOfTask(taskDone);
+        tasks.get(indexOfTask-1).markAsDone();
+        System.out.println("Nice! I've marked this task as done: ");
+        System.out.println(tasks.get(indexOfTask-1).getTask());
+    }
+
+    // Since I find that codes for tasksDone and removeTasks can be duplicated, so I refactor the common part to this function
```

Comment:
Avoid committing these kind of comments. Comments should not be used to remind yourself of something but rather be used to describe something